platform/pulpcore/app/viewsets/content.py

```diff
@@ -20,8 +23,34 @@ class Meta:
         fields = ['type']
 
 
+class ArtifactFilter(filterset.FilterSet):
+    """
+    Artifact filter Plugin content filters would need:
+     - to inherit from this class
+     - to add any plugin-specific filters if needed
+     - to define its own `Meta` class which needs:
+
+       - to specify plugin content model
+       - to extend `fields` with plugin-specific ones
+    """
+    class Meta:
+        model = Artifact
+        fields = ['md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512']
+
+
 class ContentViewSet(NamedModelViewSet):
     endpoint_name = 'content'
     queryset = Content.objects.all()
     serializer_class = ContentSerializer
     filter_class = ContentFilter
+
+
+class ArtifactViewSet(NamedModelViewSet):
+    endpoint_name = 'artifacts'
+    lookup_field = 'pk'
```

Comment:
This `lookup_field` value is a default for `NamedModelViewset`, so it's not needed.
[Docstring for `NamedModelViewset`](https://github.com/pulp/pulp/blob/3.0-dev/platform/pulpcore/app/viewsets/base.py#L16) can be fixed though since it says about the old, no longer valid, default value. My bad.